pkg/build/pipelines/git-checkout.yaml

```diff
@@ -39,65 +39,108 @@ inputs:
-        echo "Warning (git-checkout): you have not specified a branch or tag."
-      fi
-
-      git_clone_flags=""
-      if [ "${{inputs.recurse-submodules}}" == "true" ]; then
-        git_clone_flags="--recurse-submodules"
-      fi
-
-      [ -n '${{inputs.branch}}' ] && clone_target='--branch ${{inputs.branch}}'
-      [ -n '${{inputs.tag}}' ] && clone_target='--branch ${{inputs.tag}}'
-
-      workdir=$(mktemp -d)
-      mkdir -p '${{inputs.destination}}'
-      clone_fullpath=$(realpath '${{inputs.destination}}')
-
-      git config --global --add safe.directory $workdir
-      git config --global --add safe.directory $clone_fullpath
-      if [ "${{inputs.depth}}" == "-1" ]; then
-        git clone $git_clone_flags $clone_target '${{inputs.repository}}' $workdir
-      else
-        git clone $git_clone_flags $clone_target --depth '${{inputs.depth}}' '${{inputs.repository}}' $workdir
-      fi
-
-      cd $workdir
-      tar -c . | (cd $clone_fullpath && tar -x)
-      rm -rf $workdir
-      cd $clone_fullpath
-      git config --global --add safe.directory $clone_fullpath
-
-      if [ -z "${{inputs.expected-commit}}" ]; then
-        echo "Warning (git-checkout): no expected-commit"
-      elif [ -n '${{inputs.branch}}' ]; then
-        git config --global advice.detachedHead false
-        remote_commit=$(git rev-parse --verify --end-of-options "refs/heads/${{inputs.branch}}")
-        if [[ '${{inputs.expected-commit}}' != "$remote_commit" ]]; then
-          echo "Error (git-checkout): expect commit ${{inputs.expected-commit}}, got $remote_commit"
-          exit 1
-        fi
-      elif [ -n '${{inputs.tag}}' ]; then
-        # If it's a tag, then it could be a lightweight or annotated tag.
-        # Lightweight tags point directly to the commit and do not have any messages, signatures, or other data.
-        # Annotated tags point to its own git object containing the tag data, with a reference to the underlying commit.
-        # We expect most tags to be using annotated tags.
-
-        # Compare direct tag value
-        remote_commit=$(git rev-parse --verify --end-of-options "refs/tags/${{inputs.tag}}")
-        if [[ '${{inputs.expected-commit}}' == "$remote_commit" ]]; then
-          exit 0
-        fi
-
-        # Try to unpeel the tag and compare the underlying value.
-        echo "Warning (git-checkout): expected commit ${{inputs.expected-commit}}, does not match tag ${remote_commit}. Attempting to unpeel tag."
-
-        unpeeled_commit=$(git rev-parse --verify --end-of-options "refs/tags/${{inputs.tag}}^{}")
-        if [[ '${{inputs.expected-commit}}' != "${unpeeled_commit}" ]]; then
-          echo "Error (git-checkout): expect commit ${{inputs.expected-commit}}, got ${unpeeled_commit}"
-          exit 1
-        fi
-      else
-        echo "Error (git-checkout): no branch or tag provided"
-      fi
+      #!/bin/bash
+      set -e
+      set -o pipefail
```

Comment:
I just realized `set -o pipefail` and `#!/bin/bash`. I think the only bashism here is `pipefail`.

The previous version of the script had several bashism  (`[[`) but did not declare bash anywhere).
That is interesting.  are pipelines allowed to assume bash?

Reply:
Jon pointed me at https://github.com/chainguard-dev/melange/blob/ee5792a7313b7d062c3cf5960fbeaaf90350940c/pkg/build/pipeline.go#L325-L332 .

So the answer to my question is 'no', they cannot assume bash.
The interesting fact is though, that in almost all scenarioes (/bin/sh is busybox, and busybox's /bin/sh implements both `[[` and `set -o pipefail`).  So we're lucky that the previous version works, if it was ever run with dash it would not.

I went ahead and pushed a commit to drop pipefail and still check the rc value of the left side of the `tar | tar`